sdk/include/opentelemetry/sdk/metrics/view/view_registry.h

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <unordered_map>
+#  include "opentelemetry/sdk/instrumentationlibrary/instrumentation_library.h"
+#  include "opentelemetry/sdk/metrics/view/instrument_selector.h"
+#  include "opentelemetry/sdk/metrics/view/meter_selector.h"
+#  include "opentelemetry/sdk/metrics/view/view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+const std::string kDefaultViewName        = "default_name";
+const std::string kDefaultViewDescription = "default view";
+struct RegisteredView
+{
+  RegisteredView(
+      std::unique_ptr<opentelemetry::sdk::metrics::InstrumentSelector> instrument_selector,
+      std::unique_ptr<opentelemetry::sdk::metrics::MeterSelector> meter_selector,
+      std::unique_ptr<opentelemetry::sdk::metrics::View> view)
+      : instrument_selector_{std::move(instrument_selector)},
+        meter_selector_{std::move(meter_selector)},
+        view_{std::move(view)}
+  {}
+  std::unique_ptr<opentelemetry::sdk::metrics::InstrumentSelector> instrument_selector_;
+  std::unique_ptr<opentelemetry::sdk::metrics::MeterSelector> meter_selector_;
+  std::unique_ptr<opentelemetry::sdk::metrics::View> view_;
+};
+
+class ViewRegistry
+{
+public:
+  void AddView(std::unique_ptr<opentelemetry::sdk::metrics::InstrumentSelector> instrument_selector,
+               std::unique_ptr<opentelemetry::sdk::metrics::MeterSelector> meter_selector,
+               std::unique_ptr<opentelemetry::sdk::metrics::View> view)
+  {
+    // TBD - thread-safe ?
+
+    auto registered_view = std::unique_ptr<RegisteredView>(new RegisteredView{
+        std::move(instrument_selector), std::move(meter_selector), std::move(view)});
+    registered_views_.push_back(std::move(registered_view));
+  }
+
+  bool FindViews(const opentelemetry::sdk::metrics::InstrumentDescriptor &instrument_descriptor,
+                 const opentelemetry::sdk::instrumentationlibrary::InstrumentationLibrary
+                     &instrumentation_library,
+                 nostd::function_ref<bool(const View &)> callback) const
+  {
+    bool found = false;
+    for (auto const &registered_view : registered_views_)
+    {
+      if (MatchMeter(registered_view->meter_selector_.get(), instrumentation_library) &&
+          MatchInstrument(registered_view->instrument_selector_.get(), instrument_descriptor))
+      {
+        found = true;
+        if (!callback(*(registered_view->view_.get())))
+        {
+          return false;
+        }
+      }
+    }
+    // return default view if none found;
+    if (!found)
+    {
+      static View view(kDefaultViewName, kDefaultViewDescription);
```

Comment:
[blocking] The default name and description *should* come from the instrument descriptor's name/description.

Reply:
Should we return default view ( with empty name and description ) as done in Java [here](https://github.com/open-telemetry/opentelemetry-java/blob/32f7138131d18b5f1d13667ae31d35dc015d21ed/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/ViewRegistry.java#L55). This will allow us to return a default `static` immutable object always when a view is not found.

Reply:
Have made changes as per my suggestion above. Let me know if that works.

Reply:
Ah, I see, you're planning to *then* select the name of the metric by checking for empty description or empty name.

I understand wanting to keep this static.  You'll need to *either* use Java's "MetricDescriptor" trick to figure out what to name the underlying metric, or keep this logic elsewhere.

That's entirely reasonable to do, but remember you'll still be instantiating something to remember the name/description of the resulting metric.   For C++ it may be easier to do so view the View object, but you can decide once you work on the "storage" aspect of views :)